packages/js/src/base-module.ts

```diff
@@ -22,36 +22,36 @@ export class BaseModule {
 
   constructor() {
     this._emitter = NovuEventEmitter.getInstance();
-    this._apiService = ApiServiceSingleton.getInstance();
     this._inboxService = InboxServiceSingleton.getInstance();
-    this._emitter.on('session.initialize.success', ({ result }) => {
-      this.onSessionSuccess(result);
-      this.#callsQueue.forEach(async ({ fn, resolve }) => {
-        resolve(await fn());
-      });
-      this.#callsQueue = [];
-    });
-    this._emitter.on('session.initialize.error', ({ error }) => {
-      this.onSessionError(error);
-      this.#sessionError = error;
-      this.#callsQueue.forEach(({ reject }) => {
-        reject(error);
-      });
-      this.#callsQueue = [];
+    this._emitter.on('session.initialize.resolved', ({ error, data }) => {
+      if (data) {
+        this.onSessionSuccess(data);
+        this.#callsQueue.forEach(async ({ fn, resolve }) => {
+          resolve(await fn());
+        });
+        this.#callsQueue = [];
+      } else if (error) {
+        this.onSessionError(error);
+        this.#sessionError = error;
+        this.#callsQueue.forEach(({ resolve }) => {
+          resolve({ error: new NovuError('Failed to initialize session', error) });
+        });
+        this.#callsQueue = [];
+      }
```

Comment:
```else {
  throw new Error("Unknown state, contact support")
}```

Or something like that.

Reply:
This is the specific case when the session fails to initialize, I'll update it to: `Failed to initialize session, please contact the support`